Content.Client/Movement/Systems/WaddleAnimationSystem.cs

```diff
@@ -57,19 +64,29 @@ private void OnMovementInput(EntityUid entity, WaddleAnimationComponent componen
     private void OnStartedWalking(EntityUid uid, WaddleAnimationComponent component, StartedWaddlingEvent args)
     {
         if (_animation.HasRunningAnimation(uid, component.KeyName))
-        {
             return;
-        }
 
         if (!TryComp<InputMoverComponent>(uid, out var mover))
-        {
             return;
-        }
 
         if (_gravity.IsWeightless(uid))
-        {
             return;
-        }
+
+        // Do nothing if affected by stun system
+        if (
+            _statusEffects.HasStatusEffect(uid, "KnockedDown") ||
+            _statusEffects.HasStatusEffect(uid, "Stun") ||
+            _statusEffects.HasStatusEffect(uid, "SlowedDown")
+        )
+            return;
```

Comment:
It'd probably be more appropriate to opt for a `ChangeDirectionAttemptEvent` check (or any sort of similar route) instead of this. Hardcoding specific status effects like this can prove to be problematic for code maintenance

Reply:
ActionBlocker::CanMove(EntityUid) seems to work well for checking this